2.cellprofiler_processing/cputils.py

```diff
@@ -8,18 +8,22 @@
 from pathlib import Path
 
 
-def run_cellprofiler(path_to_pipeline: str, path_to_output: str, path_to_images: str):
+def run_cellprofiler(path_to_pipeline: str, path_to_output: str, path_to_images: str, plate_name: str):
     """Profile batch with CellProfiler (runs segmentation and feature extraction)
 
     Args:
         path_to_pipeline (str): path to the CellProfiler .cppipe file with the segmentation and feature measurement modules
         path_to_output (str): path to the output folder for the .sqlite file
         path_to_images (str): path to folder with IC images from specific plate
     """
-    # run CellProfiler on a plate
-    command = f"cellprofiler -c -r -p {path_to_pipeline} -o {path_to_output} -i {path_to_images}"
-    os.system(command)
-
+    # runs through any files that are in the output path
+    if any(files.name.startswith(plate_name) for files in pathlib.Path(path_to_output).iterdir()):
+        print("This plate has already been analyzed!")
+        return
+    else:
+        # run CellProfiler on a plate
+        command = f"cellprofiler -c -r -p {path_to_pipeline} -o {path_to_output} -i {path_to_images}"
+        os.system(command)
 
```

Comment:
Noticing how this function is used within `2.cellprofiler_processing/example_cp.ipynb`, there may be an opportunity to run `rename_sqlite_file` within `run_cellprofiler` after `os.system(command)`. This could mean fewer lines are needed to achieve the same results within the notebook or elsewhere.

For example:
```suggestion
    rename_sqlite_file(sqlite_file_path=path_to_output, plate=plate_name)
```

Reply:
I really like this idea! I have implemented this and I will test this to confirm that it still works and does not run a plate that already has been ran.

Reply:
It works! I am not gonna rerun the feature extraction but it does know that it has already been ran so success!